packages/libs/error-stack/src/report.rs

```diff
@@ -138,33 +155,67 @@ use crate::{
 ///     # };
 ///     # Ok(()) }
 ///     # let err = fake_main().unwrap_err();
-///     # assert_eq!(err.frames().count(), 3);
 ///     # assert!(err.contains::<ConfigError>());
 ///     # assert_eq!(err.downcast_ref::<RuntimeError>(), Some(&RuntimeError::InvalidConfig(PathBuf::from("./path/to/config.file"))));
 ///     # Ok(())
 /// }
 /// ```
+///
+/// Get the attached backtrace and spantrace:
+///
+/// ```
+/// # #![cfg_attr(nightly, feature(backtrace))]
+///
+/// # #[cfg(all(not(miri), feature = "std"))] {
+/// use error_stack::{IntoReport, ResultExt, Result};
+///
+/// # #[allow(dead_code)]
+/// # fn fake_main() -> Result<String, std::io::Error> {
+/// let config_path = "./path/to/config.file";
+/// let content = std::fs::read_to_string(config_path)
+///     .into_report()
+///     .attach_printable_lazy(|| format!("Failed to read config file {config_path:?}"));
+///
+/// let content = match content {
+///     Err(err) => {
+///         # #[cfg(all(nightly, feature = "std"))]
+///         for backtrace in err.request_ref::<std::backtrace::Backtrace>() {
+///             println!("Backtrace: {backtrace}");
+///         }   
+///
+///         # #[cfg(all(nightly, feature = "spantrace"))]
+///         for spantrace in err.request_ref::<tracing_error::SpanTrace>() {
+///             println!("Spantrace: {spantrace}")
+///         }
+///
+///         return Err(err)
+///     }
+///
+///     Ok(ok) => ok
+/// };
+///
+/// # const _: &str = stringify! {
+/// ...
+/// # }; Ok(content) }
+/// # }
+/// ```
```

Comment:
Same as above re: `miri` and `--all-features` in doc tests. This, however, does not apply to `stable` vs `nightly`.